### PR TITLE
feat: update to use GetCertificate instead of GetSecret

### LINKED ISSUE #63: Default to `Certificate / Get` for Cert Chain
Based on the description of #58 and the following code 
https://github.com/Azure/notation-azure-kv/blob/f186c0fa79d73c7d21f464cdb1af7e83d3566b03/internal/cloud/keyvault.go#L170-L188

we require `Secret/Get` permission for the certificate chain even if the ca certs are provided by the plugin config `ca_certs`.

To apply the principle of [least privileged access](https://learn.microsoft.com/azure/active-directory/develop/secure-least-privileged-access), we should default to `Certificate / Get` (Get Certificate) and only attempt `Secret/Get` (Get Certificate and Secret) when the plugin config `as_secret` is toggled to `true` or `1`.

### PR DESCRIPTION
**Pull request by JeyJeyGao**
- Use GetCertificate instead of GetSecret as the default
- Optimize error message wrapping

Resolves #63 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: internal/keyvault/keyvault.go [hunk 1]
```diff
@@ -17,29 +17,27 @@ import (
...
-
 // set of auth errors
 var (
 	errMsgAuthorizerFromMI  = "authorized from Managed Identity failed, please ensure you have assigned identity to your resource"
 	errMsgAuthorizerFromCLI = "authorized from Azure CLI 2.0 failed, please ensure you have logged in using the 'az' command line tool"
 	errMsgUnknownAuthorizer = "unknown authorize method, please use a supported authorize method according to the document"
 )
 
+// authorizer is the authorize mode used by plugin
+type authorizer string
```
Comment (shizhMSFT) [inline internal/keyvault/keyvault.go:28]: Should we call it `authMode` or `authMethod`?

File: internal/signature/signer.go [hunk 2]
```diff
@@ -44,13 +43,13 @@ func Sign(ctx context.Context, req *proto.GenerateSignatureRequest) (*proto.Gene
...
 
-	// get hash and validate hash
+	// get hash name and validate hash
 	hashName, err := proto.HashAlgorithmFromKeySpec(keySpec)
 	if err != nil {
 		return nil, err
 	}
 	if hashName != req.Hash {
-		return nil, requestErr(fmt.Errorf("keySpec hash:%v mismatch request hash:%v", hashName, req.Hash))
+		return nil, fmt.Errorf("keySpec hash:%v mismatch request hash:%v", hashName, req.Hash)
```
Comment (shizhMSFT) [inline internal/signature/signer.go:52]: nit:
```suggestion
		return nil, fmt.Errorf("keySpec hash:%v mismatch request hash: %v", hashName, req.Hash)
```

File: internal/keyvault/keyvault.go (hunk 1 above)
Comment (JeyJeyGao) [reply internal/keyvault/keyvault.go]: Updated to use `authMethod`

File: internal/signature/signer.go (hunk 2 above)
Comment (JeyJeyGao) [reply internal/signature/signer.go]: Updated.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**